pkg/api/controllers/samlconfig/samlconfig.go

```diff
@@ -0,0 +1,82 @@
+package samlconfig
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rancher/rancher/pkg/auth/providers/saml"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type authProvider struct {
+	authConfigs v3.AuthConfigInterface
+}
+
+func Register(apiContext *config.ScaledContext) {
+	a := newAuthProvider(apiContext)
+	apiContext.Management.AuthConfigs("").AddHandler("authConfigController", a.sync)
+}
+
+func newAuthProvider(apiContext *config.ScaledContext) *authProvider {
+	a := &authProvider{
+		authConfigs: apiContext.Management.AuthConfigs(""),
+	}
+	return a
+}
+
+func (a *authProvider) sync(key string, config *v3.AuthConfig) error {
+	samlConfig := &v3.SamlConfig{}
+	if key == "" || config == nil {
+		return nil
+	}
+
+	if config.Name != saml.PingName {
+		return nil
+	}
+
+	if config.Annotations["configured"] != "true" {
```

Comment:
I think this annotation is going to cause issues, but im not sure what yet because I havent fully reviewed the PR.
one issue: it is never set to false. So, if the provider is disabled, this annotation will still be true.

But im not even sure if we need it anymore. I believe it was originally needed when we had the function broken into two: configureTest and testAndApply